k8s/templates/deployment.yaml

```diff
@@ -0,0 +1,68 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name     : {{ .Values.general.name }}-deployment
+  namespace: {{ .Release.Namespace }}
+  {{- with .Values.general.labels }}
+  labels:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  replicas: {{ .Values.deployment.replicas }}
+  selector:
+    {{- with .Values.general.labels }}
+    matchLabels:
+      {{- toYaml . | nindent 6 }}
+    {{- end }}
+  strategy:
+    type: {{ .Values.deployment.strategy.type }}
+    rollingUpdate:
+      maxSurge: {{ .Values.deployment.strategy.maxSurge }}
+  template:
+    metadata:
+      {{- with .Values.general.labels }}
+      labels:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+    spec:
+      containers:
+        - name: {{ .Values.deployment.webserver.name }}
+          image: {{ .Values.deployment.webserver.image }}
+          ports:
+            - containerPort: {{ .Values.deployment.webserver.containerPort }}
+              name: {{ .Values.deployment.webserver.portName }}
+              protocol: {{ .Values.deployment.webserver.protocol }}
+          volumeMounts:
+            - name: {{ .Values.deployment.webserver.volumeName }}
+              mountPath: {{ .Values.deployment.webserver.volumeMountPath }}
+        - name: {{ .Values.deployment.app.name }}
```

Comment:
we also need to add liveness and readiness probes

Reply:
@ggramal done